src/Symfony/Component/Notifier/Bridge/Lox24/Webhook/LOX24RequestParser.php

```diff
@@ -33,28 +45,165 @@ protected function getRequestMatcher(): RequestMatcherInterface
+        }
+
+        $statusCode = $data['status_code'] ?? null;
+        $dlrCode = $data['dlr_code'] ?? null;
+
+        if (null !== $dlrCode) {
+            return $this->createEventFromDlrCode($dlrCode, $data, $payload);
+        }
+        if (null !== $statusCode) {
+            return $this->createEventFromStatusCode($statusCode, $data, $payload);
+        }
+
+        throw new RejectWebhookException(406, 'Delivery event payload is malformed - missing "status_code" or "dlr_code".');
+    }
+
+    private function createEventFromDlrCode(int $dlrCode, array $data, array $payload): ?SmsEvent
+    {
+        $eventType = match ($dlrCode) {
+            0 => null,
+            1 => SmsEvent::DELIVERED,
+            2 => null,
+            4 => null,
+            8 => SmsEvent::FAILED,
+            16 => SmsEvent::FAILED,
+            default => SmsEvent::FAILED,
+        };
+
+        if (null === $eventType) {
+            return null;
         }
 
-        $code = $data['status_code'];
+        return new SmsEvent($eventType, $data['id'], $payload);
+    }
 
-        if (0 === $code) {
+    private function createEventFromStatusCode(int $statusCode, array $data, array $payload): ?SmsEvent
+    {
+        if (0 === $statusCode) {
             return null;
         }
 
-        $name = 100 === $code ? SmsEvent::DELIVERED : SmsEvent::FAILED;
+        $eventType = match ($statusCode) {
+            100 => SmsEvent::DELIVERED,
+            208, 400, 410, 2000, 3000, 5000, 6000 => SmsEvent::FAILED,
+            default => SmsEvent::FAILED,
+        };
+
+        return new SmsEvent($eventType, $data['id'], $payload);
+    }
+
+    private function createIncomingSmsEvent(array $data, array $payload): RemoteEvent
+    {
+        if (!isset($data['from'], $data['to'], $data['text'])) {
+            $missingFields = [];
+            if (!isset($data['from'])) {
+                $missingFields[] = '"from"';
+            }
+            if (!isset($data['to'])) {
+                $missingFields[] = '"to"';
+            }
+            if (!isset($data['text'])) {
+                $missingFields[] = '"text"';
+            }
+
+            throw new RejectWebhookException(406, 'Incoming SMS payload is malformed - missing required fields: '.implode(', ', $missingFields).'.');
+        }
+
+        return new RemoteEvent('sms.received', $payload['id'], $payload);
+    }
+
+    private function createEmailParsingEvent(string $eventName, array $data, array $payload): RemoteEvent
+    {
+        $eventType = 'sms.email.parsing.success' === $eventName ? 'email.parsing.success' : 'email.parsing.failed';
+
+        return new RemoteEvent($eventType, $payload['id'], $payload);
+    }
+
+    private function createSubscriptionEvent(string $eventName, array $data, array $payload): RemoteEvent
+    {
+        if (!isset($data['phone_number'])) {
+            throw new RejectWebhookException(406, 'Subscription event payload is malformed - missing "phone_number".');
```

Comment:
Not sure why you marked the previous comment as resolved - it's not. Please double check the other ones.

```suggestion
            throw new RejectWebhookException(406, 'Subscription event payload is missing "phone_number".');
```